addons/account/models/account_bank_statement.py

```diff
@@ -714,10 +457,111 @@ def _prepare_move_line_default_vals(self, counterpart_account_id=None):
+        return company_amount, journal_amount, transaction_amount
+
     # -------------------------------------------------------------------------
     # COMPUTE METHODS
     # -------------------------------------------------------------------------
 
+    def _compute_running_balance(self):
+        line_map = {}
+
+        # Integrating the journal in the queries will make the query more complex, because we have different anchor
+        # points per journal, and it should be rare to have a read on multi journal in this model
+        self.flush_model()
+        for journal in self.journal_id:
+            journal_lines = self.filtered(lambda line: line.journal_id == journal)
+            anchor_line_domain = [
+                ('is_anchor', '=', True),
+                ('journal_id', '=', journal.id),
+            ]
+
+            min_index = min(journal_lines.mapped('internal_index'))
+            max_index = max(journal_lines.mapped('internal_index'))
+            if min_index :
+                anchor_line_domain.append(('internal_index', '<=', min_index))
+
+            anchor_index = self.search(
+                domain=anchor_line_domain,
+                order='internal_index desc',
+                limit=1
+            ).internal_index or ''
+            self._cr.execute(
+                f"""SELECT id, running_balance 
+                      FROM (
+                        SELECT st_line.id as id, statement_line_balance(amount, is_anchor, anchor_value) OVER (
+                            ORDER BY internal_index 
+                            ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW
+                        ) as running_balance
+                        FROM account_bank_statement_line st_line
+                        JOIN account_move ON account_move.id = st_line.move_id
+                        WHERE 
+                            journal_id = %s
+                          AND 
+                            internal_index >= %s 
+                          AND 
+                            internal_index <= %s 
+                        ) AS running_balances
+                      WHERE id IN %s
+                """,
+                [journal.id, anchor_index, max_index, tuple(journal_lines.ids)]
+            )
+            line_map.update({r[0]: r[1] for r in self.env.cr.fetchall()})
+        for record in self:
+            record.running_balance = line_map[record.id]
+
+    @api.depends('journal_id')
+    def _compute_statement_id(self):
+        for line in self:
+            line.statement_id = self.env['account.bank.statement'].search(
+                [
+                    ('journal_id', '=', line.journal_id.id),
+                    ('state', '=', 'open'),
+                    ('line_ids', '!=', False),  # we don't want the invisible statements to pop up here
+                ],
+                limit=1)
```

Comment:
this is very unefficient... Fetch all the values for the whole recordset with a single query

Reply:
The main goal is to do it in a form view, so I changed it to onchange